src/DataLib/fendl32B_retrofit/groupr_tools.py

```diff
@@ -110,16 +112,57 @@ def write_njoy_input_file(template):
     with open('groupr.inp', 'w') as f:
         f.write(template)
 
+def insert_after_identifier(file_str, identifier, new_line):
+    """
+    Insert a new line immediately after the last occurrence of a line
+    containing the specified identifier in the file string.
+
+    Arguments:
+        file_str (str): The content of the GENDF file as a string.
+        identifier (str): The identifier to search for in the file content.
+        new_line (str): The new line to insert after the identified line.
+
+    Returns:
+        file_str (str): The (potentially) updated content of the GENDF file.
+    """
+
+    pattern = f'({re.escape(identifier)}.*?\n)'
+    matches = list(re.finditer(pattern, file_str))
+    if matches:
+        last_match = matches[-1]
+        insert_position = last_match.end()
+        file_str = (
+            file_str[:insert_position] + new_line
+            + '\n' + file_str[insert_position:]
+        )
+
+    return file_str
+
 def ensure_gendf_markers(gendf_path, matb):
     """
     Edit the GENDF files produced from an NJOY GROUPR run to include file and
-    section records that are not automatically written out to the file by
-    NJOY. Missing these records will not cause errors, but will raise
-    messages when they are read by ENDFtk, which expects these markers, so
-    this method ensures that they are present. Edits will only be made if
-    the SEND record for MF1 or the FEND record for MF3 are not present.
-    The formatting for these records can be found at:
-    https://t2.lanl.gov/nis/endf/intro06.html
+        section records that are not automatically written out to the file by
```

Comment:
Looks like a change in indentation to take care of here.

Reply:
That's actually a question I had; I think I have been formatting my docstrings like this in general. If this indentation should not be there, I can go back and change all of them, but if you would prefer that be in a different PR for the functions that have already been merged, I can go do that too.